api/models/post/util.js

```diff
@@ -26,6 +26,16 @@ export const setupNewPostAttrs = function (userId, params) {
   return Promise.resolve(attrs)
 }
 
+const updateTagFollows = (post, trxOpts) => post.load('tags', trxOpts)
+  .then(() => post.load('communities', trxOpts))
+  .then(() => TagFollow.query(q => {
+    q.whereIn('tag_id', post.relations.tags.map('id'))
+    q.whereIn('community_id', post.relations.communities.map('id'))
+  }).fetchAll(trxOpts))
+  .then(tagFollows =>
+    tagFollows.map(tagFollow =>
```

Comment:
this should be a Promise.map -- but better yet, this whole thing can be a single query. if you call `.query()` again above with no arguments, you'll drop down to the knex API, and then you can just call `.increment()`: http://knexjs.org/#Builder-increment